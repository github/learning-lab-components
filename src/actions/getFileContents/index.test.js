const getFileContents = require('./')
const mockContext = require('../../tests/mockContext')

describe('getFileContents', () => {
  let context

  beforeEach(() => {
    context = mockContext({
      pull_request: {
        head: {
          sha: 123
        }
      }
    }, {
      gitdata: {
        getTree: jest.fn(() => Promise.resolve({ data: { tree: [{ path: 'filename', sha: 123 }] } })),
        getBlob: jest.fn(() => Promise.resolve({ data: { content: Buffer.from('this is a blob').toString('base64') } }))
      }
    })
  })

  it('gets the file contents of a given file', async () => {
    const ret = await getFileContents(context, { filename: 'filename' })
    expect(ret).toMatchSnapshot()
  })

  it('gets the file contents of a given file at a given sha', async () => {
    const ret = await getFileContents(context, { filename: 'filename', sha: 456 })
    expect(context.github.gitdata.getTree.mock.calls).toMatchSnapshot()
    expect(ret).toMatchSnapshot()
  })

  it('returns false if the content does not exist', async () => {
    context.github.gitdata.getBlob.mockImplementationOnce(() => { throw new Error(404) })
    const ret = await getFileContents(context, { filename: 'filename' })
    expect(ret).toBe(false)
  })
})
