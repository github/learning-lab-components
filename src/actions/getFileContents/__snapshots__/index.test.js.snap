// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFileContents gets the file contents of a given file 1`] = `"this is a blob"`;

exports[`getFileContents gets the file contents of a given file at a given sha 1`] = `
Array [
  Array [
    Object {
      "owner": "JasonEtco",
      "recursive": 1,
      "repo": "example",
      "tree_sha": 456,
    },
  ],
]
`;

exports[`getFileContents gets the file contents of a given file at a given sha 2`] = `"this is a blob"`;
